### ISSUE TITLE
Add option to set the width of a tab character

### ISSUE THREAD

**Issue by iomintz** [dandavison/delta#16]
Delta currently cannot calculate the correct line width when there are tabs present because it doesn't know how much width the tab takes up when it is rendered.

_Originally posted by @dandavison in https://github.com/dandavison/delta/issues/14#issuecomment-537264100_

This can be resolved by adding a flag which sets the display width of a tab character, defaulting to something sensible (2, 4, or 8).

**Comment by dandavison** [comment]
Agreed, this seems like the right solution. For reference, [bat](https://github.com/sharkdp/bat) has

```
        --tabs <T>                    
            Set the tab width to T spaces. Use a width of 0 to pass tabs through directly
```

**Comment by iomintz** [comment]
In `delta`'s case, passing tabs through directly is *broken* (why is that BTW? It works fine in diff-so-fancy.) Until/unless that is fixed, something like 0 should be disallowed.

**Comment by dandavison** [comment]
> It works fine in diff-so-fancy.)

This is because diff-so-fancy doesn't have background shading extending to the terminal width. In contrast, delta applies a background color and attempts to fill the color rightwards, to the current terminal width. To do this, it needs to know two things:
(1) The width in characters of the pseudo terminal being used by the terminal emulator. This is not a problem; `delta` uses the `console` crate for this.
(2) If your terminal renders makes some characters (e.g. tabs) take up more than one character's width on the screen, it needs to take this into account also. It needs to be told this explicitly.

> In delta's case, passing tabs through directly is broken. (why is that BTW?

It's arguable that it's not broken: it's just that delta assumes that your terminal renders all characters with the same width (including tabs). There's no clean and cross-platform way for a command line application to detect the width that your terminal emulator will use to render tabs that I'm aware of.


> Until/unless that is fixed, something like 0 should be disallowed.

Yes, I think I might have disallowed it except for one thing! One of the tests that runs as part of the test suite asserts that the diff output from delta, across the entire repo history, is byte-for-byte identical to the diff output from git without delta. So I need `--tabs=0` for that test. In practice, I've found this test helpful for catching several bugs during development. So here, I think a helpful test beats the risk that someone will set it to zero and be confused by the behavior.  In #17 I added the following option:

```
--tabs <tab_width>
    The number of spaces to replace tab characters with. Use --tabs=0 to pass tab characters through directly,
    but note that in that case delta will calculate line widths assuming tabs occupy one character's width on
    the screen: if your terminal renders tabs as more than than one character wide then delta's output will look
    incorrect. [default: 4]
```
